### PR TITLE
change license to Apache v2

### PR DESCRIPTION
**Pull request by ishidawataru**
Signed-off-by: Wataru Ishida <ishida@nel-america.com>

### CODE REVIEW

**Comment by scott-emery** [comment]
Looks good to me.

**Comment by ishidawataru** [comment]
@scott-emery Thanks for your LGTM!
@mikeypiro Could you check changing the license to Apache v2 has no problem with TIP?

**Comment by scott-emery** [comment]
I had an e-mail exchange with Young Kook Bae about the licensing. Here is his response:

I spoke with the TIP legal rep on this topic.  The bottom line is that there is an exception path that the team can take to get an approval to release TAI software under Apach 2.0 licensing.  This will require the contributors to be signed up under CLA and get an approval from the TIP board. 

The other option, is to continue to release TAI under BSD licensing while following few redistribution guidelines:

- Include the copyright notice – either in the comment section or in a separate NOTICE file – in any TAI files which include SAI codes. The text should reference the relevant file by name, for example: “<<file name>> includes software licensed by Microsoft under the Apache License, Version 2.0  <<add in the full notice as copied above>>.”
- Make sure to include the Notice.txt file in any distribution regardless of whether the distribution is in source or executable.  Also include the full text of the Apache License, Version 2.0 in a License.txt file that you include with any source or executable distribution and at the end of the Third Party Notices section in related software documentation.
- If you make any changes to the Apache code you are redistributing you should indicate in the source file that changes have been made and that the changes are licensed by TIP under the BSD license.  The same modification text should be included in the NOTICE file and the Third-Party Notices section with reference to the relevant file name.

If I’m not mistaken, most, if not everything, I listed above would still need to be done if TAI software is to be released under Apache, therefore I don’t feel that it would be any easier.

**Comment by scott-emery** [comment]
Given Young's response, it sounds like the best way to go is to use the BSD license for TAI. We need to:

- Slightly alter the copyright notices for Microsoft
- Add a NOTICE.txt file
- Change the license comments in each file

**Comment by ishidawataru** [comment]
@scott-emery Thanks for the information.
Is it hard to get an approval from the TIP board?

In my understanding, Apache v2 is appropriate for enterprise use since it has a grant of patent license clausure compared to BSD.
I heard that it is not rare to have patent issues in optical transport industries.
Since TAI is already used in enterprise user, I'd like to mitigate the risk for the users as much as we can.

**Comment by youngkbae** [comment]
Wataru-san, based on the feedback I received from the TIP legal counsel, migrating to Apache 2.0 licensing does not necessarily mitigate the risks completely.  Furthermore, the current TIP's IPR policy mandates that codes are released under the BSD or a copyright-only license (which the Apache 2.0 is not).  If we go forward with allowing **current** contributors to release under Apache 2.0, future new contributors will definitely notice the inconsistency of TAI's licensing regime.  

Please see the comment I received:

It depends on who is contributing to the code base.  Many large patent holders in this space, e.g., Qualcomm, Nokia, Ericsson, etc. are reluctant to contribute to Apache projects because the royalty-free express patent license not only covers their standards essential patents but can cover unknown downstream combinations.  Those contributors would prefer a BSD license.  If they aren’t contributors, the Apache license does not necessarily impose patent licensing obligations for merely distributing the code so they will adopt and use the code.  There is a defensive termination provision in the Apache 2.0 license that could discourage patent owners from instituting patent suits against others who have adopted the code but if it is an important patent it’s unlikely to deter them from bringing suit.
 
Bottom line depends on who you want to contribute to the project.  With that said, if the team would still find the migration desirable, I can get the paperwork started.  Please advise.

**Comment by ishidawataru** [comment]
@youngkbae Sorry for the super delay. Thank you for the information! I'm not an expert on this topic, so I will follow the suggestion from the TIP legal team.

@scott-emery I'll create a PR which will

- Slightly alter the copyright notices for Microsoft
- Add a NOTICE.txt file
- Change the license comments in each file
